components/faction/commons/faction.lua

```diff
@@ -64,6 +66,36 @@ function Faction.read(faction)
 		or Faction.aliases[faction]
 end
 
+--- Parses multiple factions from input.
+-- Returns a table of the found factions short handle/identifier.
+-- Returns empty table if no input is specified.
+---@props props {input: string, sep: string?}
+---@return table
+function Faction.readMultiFaction(props)
+	if String.isEmpty(props.input) then
+		return {}
+	end
+
+	if Faction.read(props.input) then
+		return {Faction.read(props.input)}
+	end
+
+	local input = {}
+	if String.isNotEmpty(props.sep) then
+		input = Array.map(mw.text.split(props.input, props.sep, true), mw.text.trim)
+	else
+		for char in props.input:gmatch('(.)') do
+			table.insert(input, char)
+		end
+	end
+
+	return Array.map(input, function(factionInput)
+		local faction = Faction.read(factionInput)
+		assert(faction, 'Invalid faction "' .. factionInput .. '"')
+		return faction
+	end)
```

Comment:
I think it's better to throw the whole input instead of just the first faction part that's invalid. Makes it easier to locate it in wikicode.

```suggestion
	local factions = Array.collect(input, Faction.read(factionInput))
	assert(#factions == #input, 'Invalid multi-faction specifier ' .. props.input)
	return factions
```

Reply:
Array.collect doesn't exist

```suggestion
	local factions = Array.map(input, Faction.read)
	assert(#factions == #input, 'Invalid multi-faction specifier ' .. props.input)
	return factions
```

Reply:
Oops, `Array.map` then

I added `Array.collect` in the downstream module at some point https://liquipedia.net/commons/Module:Array/downstream